classes/core/AltapaySettings.php

```diff
@@ -52,13 +54,26 @@ public function altapayOrderStatusCompleted( $orderID ) {
 		$order = new WC_Order( $orderID );
 		$txnID = $order->get_transaction_id();
 
+		$subscription = false;
+		$authType     = 'payment';
+		if ( class_exists( 'WC_Subscriptions_Order' ) && ( wcs_order_contains_subscription( $orderID, 'parent' ) || wcs_order_contains_subscription( $orderID, 'renewal' ) ) ) {
+
+			if ( $order->get_total() == 0 ) {
+				return;
+			}
+
+			$txnID        = $order->get_meta( '_agreement_id' );
+			$subscription = true;
+			$authType     = 'subscription_payment';
```

Comment:
Is this correct?

The `authType` is normally `subscription`

![Uploading image.png…]()


https://testgateway.altapaysecure.com/admin/orders/paymentDetails/ffdd7a95-2772-41ba-a473-ef9cb5628ad4

Reply:
@emicha This is not the request type, I am filtering the transaction based on the `authType` after `/merchant/API/chargeSubscription`. Since` /merchant/API/chargeSubscription` returns two transactions in response to I am identifying the transaction with `authType` as `subscription_payment`


![image](https://github.com/user-attachments/assets/f22f440d-2362-4a18-91cd-e58cef3a9fc8)